### ISSUE TITLE
Add tech for fast pause abuse on room entry

### ISSUE THREAD

**Issue by blkerby** [vg-json-data/sm-json-data#990]
It is possible to enter Bat Cave or Purple Farming Room with 1 energy and reach the farm and survive. This could come up in a scenario where getting drained by the Baby leads to one of these rooms (without any other farming opportunity).

It's also possible to reach the item on the right of Speed Booster Hall or Green Bubbles Missile Room (where, if the item is an ETank, you could survive).

I'm not sure what the worst-case frame window is for this trick, but at least it is usually more than one frame since heat damage happens only once every 4 frames. It feels like a viable trick for Extreme probably, or maybe Insane.

See videos by insomniaspeed: https://discord.com/channels/1053421401354285129/1053434704180805744/1135247641635209297

**Comment by blkerby** [comment]
Info from bobbob: entering the room with 1 energy, the frame window ranges from 2 frames in the worst case to 5 frames in the best case. 

I don't think we would model the case of entering with 0 energy (by pausing into the transition in the previous room) but in that case it would be frame perfect.

**Comment by osse101** [comment]
This would begin by better defining how much energy is saved by a normal `canPauseAbuse`, or what kind of frame window that is allowing.  I was trying it with 3 energy from the door in purple farm and that numbers feels a bit low.  When chain pause abusing to trickle in reserve energy, Its hard to accurately give yourself a low health number after fade out is complete and you can pause again.  So maybe 5 energy / 20 frames / third of a second.

Heat frames aren't accurate to <15 frames so this option doesn't seem important.  It would stand out more after the resource multiplier is applied.

Trying to add a new tech for this situation looks very similar to wanting a tech for frame perfect pausing when health reaches 0.  Or a modifier tech for `canSomeTech` but a little harder, like how difficulty presets assume you can do things a little bit better at harder tiers.  Or `canFramePreciseInputs`.

`canPauseAbuse` suggests by the name that it will be expanded to include pausing to ignore most all of a  `enemyDamage` hit.  That and pausing to use reserves to remove iframes still need to be added.

**Comment by kjbranch** [comment]
It looks like nearly all of this has been completed. Another issue could be opened for frame perfect pause abuse or pausing to use reserves to remove i-frames, if we ever decide to go down those routes.
